### PR TITLE
Update HDFS default version and remove reference to 1.x

### PR DESCRIPTION
**Pull request by ns1123**
[skip ci]

### CODE REVIEW

File: docs/en/ufs/HDFS.md [hunk 1]
```diff
@@ -31,8 +31,8 @@ directory:
 $ mvn install -P<YOUR_HADOOP_PROFILE> -D<HADOOP_VERSION> -DskipTests
 ```
 
-Alluxio provides predefined build profiles including `hadoop-1`, `hadoop-2` (enabled by default),
```
Comment (apc999) [inline docs/en/ufs/HDFS.md]: these are still true, right? are we also removing the build profile for hadoop 1?

File: docs/en/ufs/HDFS.md [hunk 2]
```diff
@@ -240,14 +240,16 @@ following line to the site properties file (`conf/alluxio-site.properties`)
...
 ```
 
 #### Supported HDFS Versions
 
 Alluxio v{{site.ALLUXIO_RELEASED_VERSION}} supports the following versions of HDFS as a valid argument of mount option `alluxio.underfs.version`:
 
-- Apache Hadoop: 1.0, 1.2, 2.2, 2.3, 2.4, 2.5, 2.6, 2.7, 2.8, 2.9, 3.0, 3.1
+- Apache Hadoop: 2.2, 2.3, 2.4, 2.5, 2.6, 2.7, 2.8, 2.9, 3.0, 3.1
+
+Note: Apache Hadoop 1.0 and 1.2 are still supported, but not included in the default download.
```
Comment (calvinjia) [inline docs/en/ufs/HDFS.md]: Include a pointer to how they can build this themselves?

File: docs/en/ufs/HDFS.md (hunk 1 above)
Comment (ns1123) [reply docs/en/ufs/HDFS.md]: I'll raise a separate PR to remove the hadoop-1 profile

Comment (apc999) [reply docs/en/ufs/HDFS.md]: oh shall we remove them from alluxio codebase? I think we can still keep them here.

Comment (calvinjia) [reply docs/en/ufs/HDFS.md]: I don't think we want to remove the `hadoop-1` profile, otherwise users cannot build `hadoop-1` by themselves?

**Comment by yuzhu** [comment]
@ns1123 @ZacBlanco do we want this in the release?

**Comment by ZacBlanco** [comment]
alluxio-bot, merge this please

**Comment by ZacBlanco** [comment]
alluxio-bot, cherry-pick this to branch-2.1 please

**Pull request merged**